src/Cake.AzureDevOps/Pipelines/AzureDevOpsBuildsHelper.cs

```diff
@@ -0,0 +1,170 @@
+﻿namespace Cake.AzureDevOps.Pipelines
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Cake.Core.Annotations;
+    using Cake.Core.Diagnostics;
+    using Microsoft.TeamFoundation.Build.WebApi;
+
+    /// <summary>
+    /// Provides functions for AzureDevOps builds.
+    /// </summary>
+    internal static class AzureDevOpsBuildsHelper
+    {
+        /// <summary>
+        /// Gets the builds for the parameter <paramref name="settings"/>.
+        /// </summary>
+        /// <param name="log">The Cake log context.</param>
+        /// <param name="settings">Settings for getting the build.</param>
+        /// <returns>The builds or an empty list of builds.</returns>
+        [CakeMethodAlias]
+        [CakeAliasCategory("Azure Pipelines")]
+        [CakeNamespaceImport("Cake.AzureDevOps.Pipelines")]
+        internal static IEnumerable<AzureDevOpsBuild> GetAzureDevOpsBuilds(
+            ICakeLog log,
+            AzureDevOpsBuildsSettings settings)
+        {
+            log.NotNull(nameof(log));
+            settings.NotNull(nameof(settings));
+
+            List<AzureDevOpsBuild> azureDevOpsBuilds = new List<AzureDevOpsBuild>();
+
+            using (var buildHttpClient = new BuildClientFactory().CreateBuildClient(settings.CollectionUrl, settings.Credentials, out var authorizedIdenity))
+            {
+                log.Verbose(
+                     "Authorized Identity:\n  Id: {0}\n  DisplayName: {1}",
+                     authorizedIdenity.Id,
+                     authorizedIdenity.DisplayName);
+
+                BuildDefinitionReference buildDefinition = null;
+
+                if (!string.IsNullOrEmpty(settings.BuildDefinitionName))
+                {
+                    buildDefinition = GetBuildDefinition(log, buildHttpClient, settings);
+                    if (buildDefinition == null)
```

Comment:
What's the idea behind returning an empty list in case a build definition is passed by settings but could not be found, instead of throwing?

Reply:
throwing would be better here, the caller knows then that the build definiton is missing and can handle better the code which calls the alias